bodhi/server/util.py

```diff
@@ -167,19 +167,32 @@ def __get__(self, obj, objtype):
 
 
 @memoized
-def get_critpath_pkgs(collection='master'):
-    """Return a list of critical path packages for a given collection"""
-    critpath_pkgs = []
+def get_critpath_components(collection='master', component_type='rpm'):
+    """ Return a list of critical path packages for a given collection.
+    :param collection: a string representing the collection/branch to search.
+    for
+    :param component_type: a string representing the component type to search
+    for. This only affects PDC queries.
+    :return: a list of critpath components
+    """
+    critpath_components = []
     critpath_type = config.get('critpath.type')
+    if critpath_type != 'pdc' and component_type != 'rpm':
+        log.warning('The critpath.type of "{0}" does not support searching for'
+                    ' non-RPM components'.format(critpath_type))
+
     if critpath_type == 'pkgdb':
         from pkgdb2client import PkgDB
         pkgdb = PkgDB(config.get('pkgdb_url'))
         results = pkgdb.get_critpath_packages(branches=collection)
         if collection in results['pkgs']:
-            critpath_pkgs = results['pkgs'][collection]
+            critpath_components = results['pkgs'][collection]
```

Comment:
I added test coverage for it but I think this one was unfair. I just renamed the variable 😛